clouddriver-artifacts/src/main/java/com/netflix/spinnaker/clouddriver/artifacts/jar/JarArtifactCredentials.java

```diff
@@ -0,0 +1,118 @@
+import org.eclipse.aether.DefaultRepositorySystemSession;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.internal.impl.DefaultRepositoryLayoutProvider;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.repository.RemoteRepository;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.resolution.ArtifactResult;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.layout.RepositoryLayoutProvider;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.file.FileTransporterFactory;
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+@Slf4j
+@Data
+public class JarArtifactCredentials implements ArtifactCredentials {
+  private final String name;
+  private final List<String> types = Collections.singletonList("jar/archive");
+
+  private final RepositorySystem repositorySystem;
+  private final RepositorySystemSession repositorySystemSession;
+  private final RemoteRepository remoteRepository;
+
+  public JarArtifactCredentials(JarArtifactAccount account) {
+    this.name = account.getName();
+    this.repositorySystem = getRepositorySystem();
+    this.repositorySystemSession = getRepositorySystemSession(this.repositorySystem);
+    this.remoteRepository = new RemoteRepository.Builder(account.getName(), "default", account.getRepositoryUrl()).build();
+  }
+
+  public InputStream download(Artifact artifact) {
+    File jarFile = getJarFile(artifact.getReference());
+    try {
+      return new FileInputStream(jarFile);
+    } catch (FileNotFoundException e) {
+      throw new IllegalStateException("File not found", e);
+    }
+  }
+
+  private RepositorySystem getRepositorySystem() {
+    DefaultServiceLocator locator = MavenRepositorySystemUtils.newServiceLocator();
+    locator.addService(RepositoryConnectorFactory.class, BasicRepositoryConnectorFactory.class);
+    locator.addService(TransporterFactory.class, FileTransporterFactory.class);
+    locator.addService(TransporterFactory.class, HttpTransporterFactory.class);
+    locator.addService(RepositoryLayoutProvider.class, DefaultRepositoryLayoutProvider.class);
+    return locator.getService(RepositorySystem.class);
+  }
+
+  private Path getRepositoryPath() throws IOException {
+    String uuid = UUID.randomUUID().toString();
+    Path tempDir = Paths.get(System.getProperty("java.io.tmpdir"), uuid);
```

Comment:
Maybe we should be sure to delete the downloaded artifact after we stream it? I can imagine this filling up pretty quickly, and generally this is only cleared after system restarts.

Reply:
It also mirrors the repo with metadata, etc...so I don't think that I can delete the artifact itself without having things end up in an inconsistent state.  I wonder if maybe an option is to have each fetch request use its own local repo, and delete it when it's done.

I wish there were a simpler API to just grab a single artifact remotely without copying it locally, but as far as I can tell that's not an option.  Let me think about this some more.

Reply:
We can leave this as-is and point it out in the docs for now.

Reply:
This will be solved if we move away from aether, which we will have to do anyway.